CDS Auto-Eject

zainmfjavaid opened the pull request.

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

Ariya3 committed:
created a toggle all function

Ariya3 committed:
created a toggle all function

spide-r committed:
testing

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

spide-r committed:
Merge branch 'CDS' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into CDS

spide-r committed:
testing

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

Ariya3 committed:
auto eject command

Ariya3 committed:
eject command

Ariya3 committed:
default command for auto eject

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

ModBoyex committed:
testing

ModBoyex committed:
testing

Ariya3 committed:
deleted a command

Ariya3 committed:
Merge branch 'CDS' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into CDS

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

Ariya3 committed:
ball color eject

ModBoyex committed:
added clarity to auto-eject

ModBoyex committed:
git fire

azvanderpas commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -20,12 +20,12 @@
-    ClimbSubsystem    (true);
+    ShooterSubsystem  (false),
+    LimelightSubsystem(false),
+    ClimbSubsystem    (false);
```
Don't forget to revert before merging

azvanderpas commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -167,15 +167,16 @@ public AimModes previous() {
+  public static final double CDSWheelControllerSpeed = 0.25;
   public static final Port colorSensorPort = Port.kOnboard; // Placeholder Value, to be changed
-  public static final int frontSensorActivation = 300;
+  public static final int frontSensorActivation = 200;
```
This value seems odd for the sensor range.

sarango2158 replied:
Is this potentially something we could tie to Shuffleboard to change on the fly as we test?

Ariya3 committed:
Co-authored-by: Noga Rosenberg <NogaRoze@users.noreply.github.com>

azvanderpas commented on src/main/java/frc/robot/commands/CDSAutoAdvanceCommand.java:
```diff
@@ -27,32 +27,31 @@ public void initialize() {}
-      if (sensorStatus[2]) {  //1 means sensor is activated
+      if (sensorStatus[2]) {  
         int nextOpenSensor = CDSSubsystem.getNextOpenSensor(sensorStatus);
         SmartDashboard.putNumber("Setpoint", nextOpenSensor);
```
Lets make this label more specific (or create a tab for CDS specifically). Make sure to update it everywhere you use it

azvanderpas commented on src/main/java/frc/robot/commands/CDSBallManagementCommand.java:
```diff
@@ -0,0 +1,104 @@
+          ejectRunning = false;
+          cycleCount = 0;
+        } else {
+          cycleCount+=60;
```
Modify msDelay to match the milliseconds you want to wait, and then increment count by 20 ms (our cycle interval)

azvanderpas commented on src/main/java/frc/robot/commands/CDSBallManagementCommand.java:
```diff
@@ -0,0 +1,104 @@
+  @Override
+  public void execute() {
+    // Run eject before auto advance
+    int ballCount = (int) SmartDashboard.getNumber("Ball Count", 0);
```
I'd rather get ballcount from CDSSubsystem instead of SmartDashboard. 

It also seems like this is meant to get ballcount from the previous getSensorStatus run, so maybe name the variable something like 'lastBallCount'?

azvanderpas commented on src/main/java/frc/robot/commands/CDSBallManagementCommand.java:
```diff
@@ -0,0 +1,104 @@
+
+
+    if (!ejectRunning) {
+      if ((ballCount > 2) || (sensorStatus[2] && CDSSubsystem.getAllianceColor() != CDSSubsystem.senseColor() && !Constants.testMode)) {
```
Long if condition deserves a comment to explain at a high level

azvanderpas commented on src/main/java/frc/robot/commands/CDSBallManagementCommand.java:
```diff
@@ -0,0 +1,104 @@
+    SmartDashboard.putBoolean("Back Sensor Status", sensorStatus[0]);
+
+
+    if (!ejectRunning) {
```
Good work on figuring this command out. At some point later we can go over how to write this as a 'state machine'.

zainmfjavaid replied:
Sounds good!

sarango2158 commented on src/main/java/frc/robot/Constants.java:
```diff
@@ -167,15 +167,15 @@ public AimModes previous() {
   public static final int middleSensorActivation = 450;
   public static final int backSensorActivation = 600;
 
+  public static final boolean testMode = true; // if false CDS will eject balls of wrong color 
```
Another possibility for tying to Shuffleboard.

sarango2158 commented on src/main/java/frc/robot/subsystems/CDSSubsystem.java:
```diff
@@ -57,18 +54,39 @@ public CDSSubsystem() {
+      SmartDashboard.putNumber("CDS Wheel Speed", -Constants.CDSWheelControllerSpeed);
+
+      CDSBeltController.getSparkMax().set(-Constants.CDSBeltSpeed);
+      CDSBeltController.setIdleMode(IdleMode.kBrake);
```
We don't need to set this idle mode over and over -- let's save time and set it once when the MotorController is created.

sarango2158 commented on src/main/java/frc/robot/subsystems/Tabs/TabContainer.java:
```diff
@@ -8,11 +8,12 @@
 
   public TabContainer(DriveBaseSubsystem d) {
-    mTabDriveBase = new TabDriveBase(d);
+    if (d != null) {
+      mTabDriveBase = new TabDriveBase(d);
+    }
```
I think the drive base team needs to come up with a more comprehensive solution here. Unfortunately this won't prevent the null reference issues in Robot.java

spide-r committed:
testing

zainmfjavaid committed:
PR fixes

zainmfjavaid commented:
Issue #135 addresses comments about shuffleboard linking for activation and test mode values. We'll be sure to work on that, but for now, we should probably focus on getting everything merged and working in main before we add new functionality.

Ariya3 committed:
Co-authored-by: Asher123456789 <Asher123456789@users.noreply.github.com>

Ariya3 committed:
Merge branch 'main' of https://github.com/AusTINCANsProgrammingTeam/2022RobotCode into CDS

Co-authored-by: Asher123456789 <Asher123456789@users.noreply.github.com>

Ariya3 committed:
cycle count

Ariya3 committed:
drive base sim corrections

Ariya3 committed:
spotless
